### PR TITLE
Node pools azure docs

### PR DESCRIPTION
**Pull request by MartaD**
First changes in the documentation

### CODE REVIEW

File: src/content/basics/nodepools/index.md [hunk 1]
```diff
@@ -90,17 +90,19 @@ spec:
...
+*For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.*
 
-Another example: In a case where you have different node pools using different availability zones. With a `nodeSelector` using the label `topology.kubernetes.io/zone` (or in Kubernetes before v1.17: `failure-domain.beta.kubernetes.io/zone`) you can assign your workload to the nodes in a particular availability zone.
+*Another example: In a case where you have different node pools using different availability zones. With a `nodeSelector` using the label `topology.kubernetes.io/zone` (or in Kubernetes before v1.17: `failure-domain.beta.kubernetes.io/zone`) you can assign your workload to the nodes in a particular availability zone.*
+
+**Is this correct for Azure as well?**
```
Comment (whites11) [inline src/content/basics/nodepools/index.md:97]: Confirmed we have the same labels in our nodes.

File: src/content/basics/nodepools/index.md [hunk 2]
```diff
@@ -90,17 +90,19 @@ spec:
...
 
-For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.
+*For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.*
 
-Another example: In a case where you have different node pools using different availability zones. With a `nodeSelector` using the label `topology.kubernetes.io/zone` (or in Kubernetes before v1.17: `failure-domain.beta.kubernetes.io/zone`) you can assign your workload to the nodes in a particular availability zone.
+*Another example: In a case where you have different node pools using different availability zones. With a `nodeSelector` using the label `topology.kubernetes.io/zone` (or in Kubernetes before v1.17: `failure-domain.beta.kubernetes.io/zone`) you can assign your workload to the nodes in a particular availability zone.*
```
Comment (MartaD) [inline src/content/basics/nodepools/index.md:95]:
```suggestion
Another example: In a case where you have different node pools using different availability zones. With a `nodeSelector` using the label `topology.kubernetes.io/zone` (or in Kubernetes before v1.17: `failure-domain.beta.kubernetes.io/zone`) you can assign your workload to the nodes in a particular availability zone.
```

File: src/content/basics/nodepools/index.md [hunk 3]
```diff
@@ -90,17 +90,19 @@ spec:
 You can assign workloads to node pools in a more indirect way too. This is achieved by using other node attributes which are
 specified via the node pool and which are exposed as node labels.
 
-For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.
+*For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.*
 
 Another example: In a case where you have different node pools using different availability zones. With a `nodeSelector` using the label `topology.kubernetes.io/zone` (or in Kubernetes before v1.17: `failure-domain.beta.kubernetes.io/zone`) you can assign your workload to the nodes in a particular availability zone.
 
+**Is this correct for Azure as well?**
```
Comment (MartaD) [inline src/content/basics/nodepools/index.md:97]:
```suggestion
```

File: src/content/basics/nodepools/index.md (hunk 1 above)
Comment (MartaD) [reply src/content/basics/nodepools/index.md]: Thanks!

File: src/content/basics/nodepools/index.md [hunk 4]
```diff
@@ -189,6 +163,17 @@ In case there are workloads not assigned to any node pools, the autoscaler may p
 
 ## Limitations
 
+- Clusters without worker nodes (= without node pools) cannot be considered fully functional. In order to have all
+required components scheduled, worker nodes are required. For that reason, we deactivate any monitoring and alerts for
+these clusters and don't provide any proactive support.
+
+- We also do not monitor node pools with less than three worker nodes and do not provide any proactive support for those.
+
+- When creating a new node pool, the master node of the cluster is re-created. This causes a downtime of the Kubernetes
```
Comment (whites11) [inline src/content/basics/nodepools/index.md:172]: Is this true??? Why???

File: src/content/basics/nodepools/index.md [hunk 5]
```diff
@@ -189,6 +163,17 @@ In case there are workloads not assigned to any node pools, the autoscaler may p
...
+API of a couple of minutes.
+
+### AWS specific
+
 - A node pool can have a maximum of 250 worker nodes. The architectural reason for this is that each node pool gets a `/24` IPv4 subnet assigned. However 5 IP addresses per availability zone used are not usable for worker nodes, as they are reserved for other AWS resources. Hence the limit depends on the number of availability zones used by a node pool. If the pool uses two zones, it's 245. With three zones, the limit is 240, and with four zones, it's 235.
```
Comment (whites11) [inline src/content/basics/nodepools/index.md:177]: This is not true on Azure, because we have the 100 limitation at the VMSS level. Let me rephrase that.

Comment (whites11) [reply src/content/basics/nodepools/index.md:177]: OK this was the AWS section, sorry

File: src/content/basics/nodepools/index.md [hunk 6]
```diff
@@ -90,17 +90,18 @@ spec:
...
+*For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.*
 
 Another example: In a case where you have different node pools using different availability zones. With a `nodeSelector` using the label `topology.kubernetes.io/zone` (or in Kubernetes before v1.17: `failure-domain.beta.kubernetes.io/zone`) you can assign your workload to the nodes in a particular availability zone.
 
+
 ## Node pool deletion
 
 You can delete a node pool at any time using the Giant Swarm API and user interfaces. When a node pool gets deleted the following things will happen:
 
 - nodes in the pool will be marked as unschedulable and then drained, resulting in Pods being unassigned from the nodes
```
Comment (whites11) [inline src/content/basics/nodepools/index.md:102]: This does not happen in azure currently

File: src/content/basics/nodepools/index.md (hunk 4 above)
Comment (MartaD) [reply src/content/basics/nodepools/index.md:172]: I don't know, it's different on Azure? I think the three nodes node pool is a bit extreme, we have a limit of at least 3 nodes per cluster but we still monitor and support it afaik. Should we ask Firecracker?

File: src/content/basics/nodepools/index.md (hunk 6 above)
Comment (MartaD) [reply src/content/basics/nodepools/index.md:102]: is it different process? should it be separated to "following things will happen:
on AWS: ..
on Azure: .. " ?

File: src/content/basics/nodepools/index.md (hunk 4 above)
Comment (cornelius-keller) [reply src/content/basics/nodepools/index.md:172]: Are we recrating the master node if we create a NodePool?

File: src/content/basics/nodepools/index.md [hunk 7]
```diff
@@ -87,20 +87,36 @@ spec:
...
+    image: nginx
+  nodeSelector:
+    giantswarm.io/machine-pool: a1b2c
+```
+
 You can assign workloads to node pools in a more indirect way too. This is achieved by using other node attributes which are
 specified via the node pool and which are exposed as node labels.
 
-For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.
+*For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.*
```
Comment (MartaD) [inline src/content/basics/nodepools/index.md:108]:
```suggestion
For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.
```

File: src/content/basics/nodepools/index.md (hunk 4 above)
Comment (calvix) [reply src/content/basics/nodepools/index.md:187]: this was true in the old AWS architecture, now with ASG masters we do not recreate master

File: src/content/basics/nodepools/index.md (hunk 6 above)
Comment (whites11) [reply src/content/basics/nodepools/index.md:117]: I got away with "(only on AWS clusters)"

**Review by whites11 — APPROVED** [review]

File: src/content/basics/nodepools/index.md [hunk 8]
```diff
@@ -24,7 +24,7 @@ availability zone distribution, even if some workloads wouldn't require the incr
 Node pools are independent groups of worker nodes belonging to a cluster, where all nodes within a pool share a
 common configuration. You can combine any type of node pool within one cluster. Node pools can differ regarding:
 
-- EC2 instance type
+- Instance type
```
Comment (fiunchinho) [inline src/content/basics/nodepools/index.md:27]: It's hard to explain things when cloud providers use different names for the same concept. I was thinking if it would be better to write `Instance type / VM size`?

**Review by fiunchinho — APPROVED** [review]
LGTM

File: src/content/basics/nodepools/index.md (hunk 8 above)
Comment (omegas27) [reply src/content/basics/nodepools/index.md:27]: Would Machine type encapsulate both ?

File: src/content/basics/nodepools/index.md [hunk 9]
```diff
@@ -87,20 +87,36 @@ spec:
...
 For example: In a case where you have node pools with one instance type. Using a `nodeSelector` with the label `node.kubernetes.io/instance-type` (or in Kubernetes before v1.17: `beta.kubernetes.io/instance-type`) you can assign workloads to matching nodes only.
 
 Another example: In a case where you have different node pools using different availability zones. With a `nodeSelector` using the label `topology.kubernetes.io/zone` (or in Kubernetes before v1.17: `failure-domain.beta.kubernetes.io/zone`) you can assign your workload to the nodes in a particular availability zone.
 
+
 ## Node pool deletion
 
 You can delete a node pool at any time using the Giant Swarm API and user interfaces. When a node pool gets deleted the following things will happen:
 
 - nodes in the pool will be marked as unschedulable and then drained, resulting in Pods being unassigned from the nodes
```
Comment (omegas27) [inline src/content/basics/nodepools/index.md:117]: Does this mean the nodes are node drained before you remove them? That means the will have some downtime when a node pool is deleting right?

Comment (MartaD) [reply src/content/basics/nodepools/index.md:117]: @whites11 , I need help 😁

File: src/content/basics/nodepools/index.md (hunk 8 above)
Comment (MartaD) [reply src/content/basics/nodepools/index.md:27]: +1 if you agree with `machine type` 😁

File: src/content/basics/nodepools/index.md (hunk 9 above)
Comment (whites11) [reply src/content/basics/nodepools/index.md:117]: yes @omegas27 we took the decision to not implement this feature, because we don't do it with downscaling anyway (azure lifecycle callbacks don't allow this).

File: src/content/basics/nodepools/index.md (hunk 8 above)
Comment (MartaD) [inline src/content/basics/nodepools/index.md:27]:
```suggestion
- Machine type
```
